lib/webenginesettings.cpp

```diff
@@ -191,20 +204,27 @@ void SailfishOS::WebEngineSettings::initialize()
     qreal touchStartTolerance = dragThreshold / QGuiApplication::primaryScreen()->physicalDotsPerInch();
     engineSettings->setPreference(QString("apz.touch_start_tolerance"), QString("%1f").arg(touchStartTolerance));
 
-    qreal pixelRatio = SAILFISH_WEBENGINE_DEFAULT_PIXEL_RATIO * silicaTheme->pixelRatio();
-    // Round to nearest even rounding factor
-    pixelRatio = qRound(pixelRatio / 0.5) * 0.5;
+    QSize screenResolution = QGuiApplication::primaryScreen()->size();
+    int screenWidth = screenResolution.width();
+
+    qreal pixelRatio = mappedPixelRatio(screenResolution);
+
+    // Try to calculate pixelRatio only if not mapped.
+    if (pixelRatio == -1.0) {
+        pixelRatio = SAILFISH_WEBENGINE_DEFAULT_PIXEL_RATIO * silicaTheme->pixelRatio();
+        // Round to nearest even rounding factor
+        pixelRatio = qRound(pixelRatio / 0.5) * 0.5;
```

Comment:
Side note: I don't know the requirements for a good web pixel ratio, but from the overridden value with lots of decimal I wonder if the below calculation is generally good. If there's no need to get "round" ratios (1.5, 2, 3, etc) then maybe silica ratio shouldn't be used for calculation source?